app/scripts/components/login.jsx

```diff
@@ -28,14 +29,20 @@ export default React.createClass({
     };
   },
 
-  usernameChange: function(evt) {
-    this.setState({username: evt.target.value});
-  },
+  shouldComponentUpdate: function(nextProps) {
```

Comment:
This component will only receive the update if it's render so no. You will only be redirect when you're here.